tests/suites/hooks/reboot.sh

```diff
@@ -27,17 +27,19 @@ run_start_hook_fires_after_reboot() {
     # Restart the unit agent and ensure that the implicit start hook still
     # does not fire
     echo "[+] ensuring that implicit start hook does not fire after restarting the unit agent"
-    juju run --unit ubuntu-lite/0 'sudo service jujud-unit-ubuntu-lite-0 restart'
+    # juju run --unit ubuntu-lite/0 'sudo service jujud-unit-ubuntu-lite-0 restart'
```

Comment:
I was looking into why this particular test was failing earlier today (e.g. https://jenkins.juju.canonical.com/job/test-hooks-lxd/449/console) and I noticed this error from the `juju run` on this line:

```Failed to restart jujud-unit-ubuntu-lite-0.service: Unit jujud-unit-ubuntu-lite-0.service not found.```

I believe we are hitting some kind of race here where the service file has not been written yet  even though status shows the unit as being idle. How about if we add a small helper to wait for the service to appear before attempting to restart? Something like:

```bash
# Call like: wait_for_unit_service_to_appear "ubuntu-lite/0" "ubuntu-lite-0"
wait_for_unit_service_to_appear() {
  unit=$1
  svc=$2
  for i in 1 2 3; do
    svc_present=$(juju ssh "$unit" "ls /etc/systemd/system/jujud-unit-${svc}.service || echo 'missing'")
    if [ "$svc_present" != "missing" ]; then
      juju ssh "$unit" "sudo systemctl restart jujud-unit-${svc}"
      return
    fi
 
    # Wait for service file to appear
    sleep 1
  done

   echo $(red "Systemd service file for ${svc} not present")
   exit 1
}
```

I think the upgrade-charm bit below is a nice addition for regression checks but I also believe we should also keep the above test to ensure that reboot is not incorrectly handled when the service is externally restarted.